### PR TITLE
Explainer change for supporting preferred platform for cross app web attribution

### LINKED ISSUE #1148: Provide API measurement platform preference while registering sources and triggers
Currently, Ad-techs (reporting endpoint) determines the availability of OS-level support based on the `Attribution-Reporting-Support` containing `OS` presence. 

There certain scenarios where `Attribution-Reporting-Support` is unavailable, for example pre-existing conversion tags which do not include the `attributionsrc` attribute. 

Provide a support in `Attribution-Reporting-Register-Source` and `Attribution-Reporting-Register-Trigger` for **`preferred-platform`** as:

e.g. `Attribution-Reporting-Register-Source` would look like 
```
 {
  "preferred_platform": "os" (possible values: web or os)

  "register_os_source": [ 
        { "reporting_endpoint":  "https://adtech.example/register", 
           "debug_reporting": true
        },
        { "reporting_endpoint":  "https://other-adtech.example/register", 
           "debug_reporting": true
        },
     ]

 // Already existing fields in `Attribution-Reporting-Register-Source`
 "destination": "[eTLD+1]",
  "source_event_id": "[64-bit unsigned integer]",
  "expiry": "[64-bit signed integer]",
  "priority": "[64-bit signed integer]",
}
```

Few advantages:
- This way, Ad-techs can prefer to register using one API over the other, whenever available.
- Defer the API platform choice decision to Chrome client / Privacy Sandbox API.
- This would also reduce overhead of sending yet another response header `Attribution-Reporting-Register-OS-Source`.

**Debugging Support:**  Respond with `os-source|trigger-delegated` `success` or `failure` response if the delegation to OS-level was performed or not. 

FYI, current behavior of the API seems to be when reporting endpoint returns with both `Attribution-Reporting-Register-Source|Trigger` and `Attribution-Reporting-Register-OS-Source|Trigger`, the source and trigger is rejected.

### PR DESCRIPTION
**Pull request by linnan-github**
Fixes #1148

### CODE REVIEW

**Comment by linnan-github** [comment]
@arpanah, could you PTAL, thanks!

File: app_to_web.md [hunk 1]
```diff
@@ -66,6 +66,17 @@ Trigger registrations will accept a new response header as well:
 Attribution-Reporting-Register-OS-Trigger: "https://adtech.example/register", "https://other-adtech.example/register"
 ```
 
+The reporting origin can also optionally respond with a [dictionary structured header](https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-header-structure-15#section-3.2)
+`Attribution-Reporting-Info` to specifiy the preferred platform. The key is
```
Comment (apasel422) [inline app_to_web.md:70]:
```suggestion
`Attribution-Reporting-Info` to specify the preferred platform. The key is
```

**Review by apasel422 — DISMISSED** [review]

File: app_to_web.md [hunk 2]
```diff
@@ -66,6 +66,17 @@ Trigger registrations will accept a new response header as well:
 Attribution-Reporting-Register-OS-Trigger: "https://adtech.example/register", "https://other-adtech.example/register"
 ```
 
+The reporting origin can also optionally respond with a [dictionary structured header](https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-header-structure-15#section-3.2)
+`Attribution-Reporting-Info` to specify the preferred platform. The key is
+`preferred-platform` and the value is a [structured header token](https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-header-structure-15#section-3.3.4)
```
Comment (apasel422) [inline app_to_web.md:71]: Can we avoid the draft links and instead use the same links from our spec, ie.. https://httpwg.org/specs/rfc8941.html#dictionary ?

Comment (linnan-github) [reply app_to_web.md]: Updated, thanks.

**Review by apasel422 — APPROVED** [review]

**Review by akashnadan — APPROVED** [review]

File: app_to_web.md [hunk 3]
```diff
@@ -74,7 +85,9 @@ Normal attribution logic in the browser will be halted.
 
 For a site to enable App<->Web attribution after integrating with the Web API, they will only need to make server side changes.
 
-A reporting origin responding with the `Attribution-Reporting-Register-OS-Source` or `Attribution-Reporting-Register-OS-Trigger` headers while `Attribution-Reporting-Support` does not contain `os` will cause the entire registration to fail (including web registration).
+A reporting origin responding with the `Attribution-Reporting-Register-OS-Source` or `Attribution-Reporting-Register-OS-Trigger` headers while there is no OS-level support will cause the OS registration to fail.
```
Comment (arpanah) [inline app_to_web.md:89]: Could we clarify these 2  statements

**Pull request merged**